homeassistant/components/surepetcare/__init__.py

```diff
@@ -142,18 +113,36 @@ async def handle_set_lock_state(call):
             vol.Required(ATTR_LOCK_STATE): vol.All(
                 cv.string,
                 vol.Lower,
-                vol.In(lock_states.keys()),
+                vol.In(coordinator.lock_states.keys()),
             ),
         }
     )
-
     hass.services.async_register(
         DOMAIN,
         SERVICE_SET_LOCK_STATE,
-        handle_set_lock_state,
+        coordinator.handle_set_lock_state,
         schema=lock_state_service_schema,
     )
 
+    set_pet_location_schema = vol.Schema(
+        {
+            vol.Optional(ATTR_PET_NAME): vol.In(coordinator.get_pets().values()),
+            vol.Required(ATTR_LOCATION): vol.In(
+                [
+                    Location.INSIDE.name.title(),
+                    Location.OUTSIDE.name.title(),
+                    Location.UNKNOWN.name.title(),
```

Comment:
don't use Unknown or at least don't call the API with that, that's HA-only, not from Sure Petcare